<h3 style="margin:0px">Class: org.apache.hadoop.hbase.errorhandling.TestTimeoutExceptionInjector (4 methods) </h3><br><pre style="padding:0"><code class="html"><div id="tags"><button id="1"style="border-width:0;padding:0;background:#282B2E"  onclick="filter('1')" data-toggle="tooltip" title="Executes methods or other tests from the same test unit"><kbd id="tag-1"class="label-info"style="display: inline-block;font-size:7pt" >ExecutionTester&nbsp;(2)</kbd></button>&nbsp;<button id="11"style="border-width:0;padding:0;background:#282B2E"  onclick="filter('11')" data-toggle="tooltip" title="Verifies (un)successful execution of the test case by reporting explicitly a failure"><kbd id="tag-11"class="label-info"style="display: inline-block;font-size:7pt" >UtilityVerifier&nbsp;(2)</kbd></button>&nbsp;</div></code></pre><center><button onclick="showBlocks()" style="margin:6px">Update filter <span class="glyphicon glyphicon-refresh" aria-hidden="true"></span> </button></center><br>
<pre class="type-11 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies (un)successful execution of the test case by reporting explicitly a failure">UtilityVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies (un)successful execution of the test case by reporting explicitly a failure
"></span><br>
/** 
 * Demonstrate TimeoutExceptionInjector semantics -- completion means no more exceptions passed to
 * error listener.
 */
@Test(timeout=60000) public void testStartAfterComplete() throws InterruptedException {
  final long time=10;
  ForeignExceptionListener listener=Mockito.mock(ForeignExceptionListener.class);
  TimeoutExceptionInjector timer=new TimeoutExceptionInjector(listener,time);
  timer.complete();
  try {
    timer.start();
    fail("Timer should fail to start after complete.");
  }
 catch (  IllegalStateException e) {
    LOG.debug("Correctly failed timer: " + e.getMessage());
  }
  Thread.sleep(time + 1);
  Mockito.verifyZeroInteractions(listener);
}

</code></pre>

<pre class="type-11 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies (un)successful execution of the test case by reporting explicitly a failure">UtilityVerifier</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies (un)successful execution of the test case by reporting explicitly a failure
"></span><br>
/** 
 * Demonstrate TimeoutExceptionInjector semantics -- triggering fires exception and completes
 * the timer.
 */
@Test(timeout=60000) public void testStartAfterTrigger() throws InterruptedException {
  final long time=10;
  ForeignExceptionListener listener=Mockito.mock(ForeignExceptionListener.class);
  TimeoutExceptionInjector timer=new TimeoutExceptionInjector(listener,time);
  timer.trigger();
  try {
    timer.start();
    fail("Timer should fail to start after complete.");
  }
 catch (  IllegalStateException e) {
    LOG.debug("Correctly failed timer: " + e.getMessage());
  }
  Thread.sleep(time * 2);
  Mockito.verify(listener,Mockito.times(1)).receive(Mockito.any(ForeignException.class));
  Mockito.verifyNoMoreInteractions(listener);
}

</code></pre>

<pre class="type-1 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test that a manually triggered exception with data fires with the data in receiveError.
 */
@Test public void testTimerPassesOnErrorInfo(){
  final long time=1000000;
  ForeignExceptionListener listener=Mockito.mock(ForeignExceptionListener.class);
  TimeoutExceptionInjector timer=new TimeoutExceptionInjector(listener,time);
  timer.start();
  timer.trigger();
  Mockito.verify(listener).receive(Mockito.any(ForeignException.class));
}

</code></pre>

<pre class="type-1 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Executes methods or other tests from the same test unit
"></span><br>
/** 
 * Test that a manually triggered timer fires an exception.
 */
@Test(timeout=60000) public void testTimerTrigger(){
  final long time=10000000;
  ForeignExceptionListener listener=Mockito.mock(ForeignExceptionListener.class);
  TimeoutExceptionInjector timer=new TimeoutExceptionInjector(listener,time);
  timer.start();
  timer.trigger();
  Mockito.verify(listener,Mockito.times(1)).receive(Mockito.any(ForeignException.class));
}

</code></pre>

<script>$(document).ready(function() {
  $('pre code').each(function(i, block) {
    hljs.highlightBlock(block);
  });
});</script>
